services/api/lib/postgre.js

```diff
@@ -295,6 +295,334 @@ module.exports = function (pg) {
+          case 'projects': {
+            switch (method) {
+              case 'create': {
+                values = [
+                  userId,
+                  null,
+                  server.methods.utils.version(),
+                  data.title,
+                  JSON.stringify(data.thumbnail || {})
+                ];
+                break;
+              }
+              case 'update': {
+                values = [
+                  data.title,
+                  data.id
+                ];
+                break;
+              }
+              case 'remove': {
+                values = [
+                  data.id
+                ];
+                break;
+              }
+            }
+            break;
+          }
+          case 'pages': {
+            switch (method) {
+              case 'create': {
+                values = [
+                  data.projectId,
+                  userId,
+                  data.x,
+                  data.y,
+                  JSON.stringify(data.styles || {})
+                ];
+                break;
+              }
+              case 'update': {
+                values = [
+                  data.x,
+                  data.y,
+                  JSON.stringify(data.styles || {}),
+                  data.id
+                ];
+                break;
+              }
+              case 'remove': {
+                values = [
+                  data.id
+                ];
+                break;
+              }
+            }
+            break;
+          }
+          case 'elements': {
+            switch (method) {
+              case 'create': {
+                values = [
+                  data.pageId,
+                  userId,
+                  data.type,
+                  JSON.stringify(data.attributes || {}),
+                  JSON.stringify(data.styles || {})
+                ];
+                break;
+              }
+              case 'update': {
+                values = [
+                  JSON.stringify(data.styles || {}),
+                  JSON.stringify(data.attributes || {}),
+                  data.id
+                ];
+                break;
+              }
+              case 'remove': {
+                values = [
+                  data.id
+                ];
+                break;
+              }
+            }
+          }
+        }
+        return values;
+      }
+
+      function hasPermissionsForAction(userId, type, method, data, results, transaction) {
+        return new Promise(function(resolve, reject) {
+          var lookupType;
+          var lookupId;
+
+          // determine the type of query to run, and the id value to look up
+          if ( method === 'create' ) {
+            switch (type) {
+              case 'projects': {
+                // the user should always be able to create new projects under their account, so resolve true now
+                return resolve(true);
+              }
+              case 'pages': {
```

Comment:
yes, because I need to make sure that the user owns the project or page that a page or element is being created for, otherwise someone can create pages and elements for projects they don't own. 

Adding user_id to pages and elements simplifies the update/edit checks only